mopro-core/src/middleware/circom/mod.rs

```diff
@@ -28,13 +29,21 @@ use once_cell::sync::{Lazy, OnceCell};
 use wasmer::{Module, Store};
 
 use ark_zkey::{read_arkzkey, read_arkzkey_from_bytes}; //SerializableConstraintMatrices
+use witness::{init_graph, Graph}; // use witness::Graph;
 
 #[cfg(feature = "dylib")]
 use {
     std::{env, path::Path},
     wasmer::Dylib,
 };
 
+#[cfg(feature = "calc-witness")]
```

Comment:
what is difference between build-witness and calc-witness flags?

Reply:
`build-witness` is more like in prepare stage that it build a witness graph (`graph.bin`)
but it uses `witness-circom-rs` package and it takes the circom path

and for `calc-witness` it just to use the `graph.bin` to generate witness in proof

Is there a better way to `build-witness` in prepare script(!?

Reply:
I would just gate them under the same feature flag even if it is different parts of the code? From and end user there is only one decision: should I use experimental witness stack or not. Too many options gets confusing. Is there a good reason to only enable one of these flags?